### ISSUE TITLE
`eval` breaks `no-undef` and `no-unused-vars` rules

### ISSUE THREAD

**Issue by Hypnosphi** [babel/babel-eslint#560]
`babel-eslint@8.1.1`
`eslint@4.14.0`

#### `.eslintrc`
```yml
parser: babel-eslint
env:
  browser: true
rules:
  no-undef: error
  no-unused-vars: error
```

#### `index.js`
```js
const a = 1;
console.log(a);

eval('');
```

```sh
$ eslint index.js

/Users/jetbrains/IdeaProjects/eslint-repro/index.js
  1:7   error  'a' is not defined             no-undef
  1:7   error  'a' is defined but never used  no-unused-vars
  2:13  error  'a' is not defined             no-undef
```

After commenting the `eval` line, false positives disappear

**Comment by mysticatea** [comment]
Thank you for the report.

I confirmed it. Weird.
